File: python/ittapi.native/collection_control.cpp
#include "collection_control.hpp"

#include <ittnotify.h>


namespace ittapi
{

PyObject* pause(PyObject* self, PyObject* Py_UNUSED(args))
{
    Py_BEGIN_ALLOW_THREADS;
    __itt_pause();
    Py_END_ALLOW_THREADS;
    Py_RETURN_NONE;
}

PyObject* resume(PyObject* self, PyObject* Py_UNUSED(args))
{
    Py_BEGIN_ALLOW_THREADS;
    __itt_resume();
    Py_END_ALLOW_THREADS;
    Py_RETURN_NONE;
}

PyObject* detach(PyObject* self, PyObject* Py_UNUSED(args))
{
    Py_BEGIN_ALLOW_THREADS;
    __itt_detach();
    Py_END_ALLOW_THREADS;
    Py_RETURN_NONE;
}

} // namespace ittapi

File: python/itt_NERSC/collection_control.py
from ittapi.native import detach as _detach, pause as _pause, resume as _resume

def detach():
    """
    Detach collection of profiling data.
    """
    _detach()

def resume():
    """
    Resume collection of profiling data.
    """
    _resume()

def pause():
    """
    Pause collection of profiling data.
    """
    _pause()